File: outputs/2025-08-13_14-23-46-994791/test4.py
# Unit test result: ✅ PASSED
from abc import ABC, abstractmethod

class Operacao(ABC):

    @abstractmethod
    def aplicar(self, a: int, b: int) -> int:
        pass

class Soma(Operacao):

    def aplicar(self, a: int, b: int) -> int:
        return a + b

class TesteInterface:

    @staticmethod
    def main(args: list[str]) -> None:
        op = Soma()
        print(op.aplicar(3, 4))
if __name__ == '__main__':
    TesteInterface.main([])

File: uploads/2025-08-13_14-23-46-994791/test4.java
interface Operacao {
    int aplicar(int a, int b);
}

class Soma implements Operacao {
    public int aplicar(int a, int b) {
        return a + b;
    }
}

class TesteInterface {
    public static void main(String[] args) {
        Operacao op = new Soma();
        System.out.println(op.aplicar(3, 4)); // 7
    }
}
